connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -184,6 +185,33 @@ public void resumeConnector(String connector) {
         configBackingStore.putTargetState(connector, TargetState.STARTED);
     }
 
+    @Override
+    public ConfigDef getConfigDef(String connName) {
+        Connector connector = getConnector(connName);
+        if (connector == null) {
+            throw new NotFoundException("No status found for connector " + connName);
+        }
+
+        return connector.config();
+    }
+
+    @Override
+    public Map<String, String> maskCredentials(String connName, Map<String, String> config) {
+        ConfigDef configDef = getConfigDef(connName);
+        Map<String, String> newConfig = new LinkedHashMap<>();
+
+        for (Map.Entry<String, String> entry : config.entrySet()) {
+            final String key = entry.getKey();
+            if (configDef.configKeys().containsKey(key) && configDef.configKeys().get(key).type.equals(ConfigDef.Type.PASSWORD)) {
```

Comment:
Can this be made polymorphic instead of actually checking the `PASSWORD` type here? e.g., could we simply always do `newConfig.put(key, entry.getValue().toString())` since `Password.toString()` always hides the value (in contrast to the `value()` method? I don't see a use atm, but keeping this generalizable seems potentially valuable.

wouldn't block on this, but it doesn't seem great to have to bake details of the config type into code in the `AbstractHerder` that otherwise doesn't care about it.